### ISSUE TITLE
Failed Mac OS github action build: BOOST_MPL_ASSERT

### ISSUE THREAD

**Issue by ghost** [aeonix/aeon#231]
```

In file included from /Users/runner/work/aeon/aeon/src/checkpoints/checkpoints.cpp:41:
In file included from /Users/runner/work/aeon/aeon/contrib/epee/include/storages/portable_storage_template_helper.h:32:
/Users/runner/work/aeon/aeon/contrib/epee/include/storages/portable_storage.h:213:7: error: expected expression
      BOOST_MPL_ASSERT(( boost::mpl::contains<storage_entry::types, t_value> )); 
      ^
/usr/local/include/boost/mpl/assert.hpp:300:47: note: expanded from macro 'BOOST_MPL_ASSERT'
              boost::mpl::assert_arg( (void (*) pred)0, 1 ) \
                                              ^
In file included from /Users/runner/work/aeon/aeon/src/checkpoints/checkpoints.cpp:41:
In file included from /Users/runner/work/aeon/aeon/contrib/epee/include/storages/portable_storage_template_helper.h:32:
/Users/runner/work/aeon/aeon/contrib/epee/include/storages/portable_storage.h:213:67: error: expected '(' for function-style cast or type construction
      BOOST_MPL_ASSERT(( boost::mpl::contains<storage_entry::types, t_value> )); 
                                              ~~~~~~~~~~~~~~~~~~~~^
/usr/local/include/boost/mpl/assert.hpp:300:49: note: expanded from macro 'BOOST_MPL_ASSERT'
              boost::mpl::assert_arg( (void (*) pred)0, 1 ) \
                                                ^~~~
/usr/local/include/boost/mpl/assert.hpp:60:58: note: expanded from macro '\
BOOST_MPL_AUX_ASSERT_CONSTANT'
#   define BOOST_MPL_AUX_ASSERT_CONSTANT(T, expr) enum { expr }
                                                         ^~~~
```
https://github.com/aeonix/aeon/runs/2865113437?check_suite_focus=true

**Comment by ghost** [comment]
Found relevant issue in monero: https://github.com/monero-project/monero/issues/7728

Added fix to PR #225

**Comment by mj-xmr** [comment]
Could you please try `Master` now?

**Comment by ghost** [comment]
This is for aeon project so I don't think it relates to monero. But please say otherwise if I am wrong. I never had this problem personally, it is reported in the github actions. Then I added the fix to a PR #225 I am working on related to CMake. I will report any issues if they arise though. Thanks for stopping by to check.

**Comment by ghost** [comment]
This is included in #228
